web/__init__.py

```diff
@@ -32,60 +52,77 @@
 
 @app.before_first_request
 def startup():
-    app.svc.register("pppp", web.service.pppp.PPPPService())
-    app.svc.register("videoqueue", web.service.video.VideoQueue())
-    app.svc.register("mqttqueue", web.service.mqtt.MqttQueue())
-    app.svc.register("filetransfer", web.service.filetransfer.FileTransferService())
+    if app.config["login"]:
+        app.svc.register("pppp", web.service.pppp.PPPPService())
+        app.svc.register("videoqueue", web.service.video.VideoQueue())
+        app.svc.register("mqttqueue", web.service.mqtt.MqttQueue())
+        app.svc.register("filetransfer", web.service.filetransfer.FileTransferService())
 
 
 @sock.route("/ws/mqtt")
 def mqtt(sock):
 
-    for data in app.svc.stream("mqttqueue"):
-        log.debug(f"MQTT message: {data}")
-        sock.send(json.dumps(data))
+    if app.config["login"]:
+        for data in app.svc.stream("mqttqueue"):
+            log.debug(f"MQTT message: {data}")
+            sock.send(json.dumps(data))
 
 
 @sock.route("/ws/video")
 def video(sock):
 
-    for msg in app.svc.stream("videoqueue"):
-        sock.send(msg.data)
+    if app.config["login"]:
+        for msg in app.svc.stream("videoqueue"):
+            sock.send(msg.data)
 
 
 @sock.route("/ws/ctrl")
 def ctrl(sock):
 
-    while True:
-        msg = json.loads(sock.receive())
+    if app.config["login"]:
```

Comment:
Instead of indenting everything in all of these websocket functions, let's follow this pattern instead:

```python
if not app.config["login"]:
    return
```

That way, the core logic isn't pushed needlessly to the right.